Tests fail on new SQLAlchemy

## 🐛 Bug description

Mining cache tests fail on the latest SQLAlchemy version `1.4.0`.

All our CI tests will fail util this is resolved since in tox we install via `setup.py` and not via `requirements.txt`.

## To reproduce

```bash
pip install "SQLAlchemy==1.4.0"
pytest tests/test_database/test_mining_cache.py::TestCreateMiningCache
```
Fail.

```bash
pip install "SQLAlchemy<1.4.0"
pytest tests/test_database/test_mining_cache.py::TestCreateMiningCache
```
Success.

## Details
Traceback:
```
        cache_creator_instance = CreateMiningCache(
            database_engine=fake_sqlalchemy_engine,
            ee_models_library=ee_models_library,
            target_table_name="mining_cache_temporary",
>           workers_per_model=2,
        )

tests/test_database/test_mining_cache.py:190: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/bluesearch/database/mining_cache.py:282: in __init__
    if not database_engine.dialect.has_table(database_engine, table_name):
venv/lib/python3.7/site-packages/sqlalchemy/dialects/sqlite/base.py:1999: in has_table
    connection, "table_info", table_name, schema=schema
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <sqlalchemy.dialects.sqlite.pysqlite.SQLiteDialect_pysqlite object at 0x126206750>
connection = Engine(sqlite:////private/var/folders/cz/v1lxwy5579502ksd81z14_t40000gp/T/pytest-of-sschmidt/pytest-6/db/cord19_test.db)
pragma = 'table_info', table_name = 'articles', schema = None

    def _get_table_pragma(self, connection, pragma, table_name, schema=None):
        quote = self.identifier_preparer.quote_identifier
        if schema is not None:
            statements = ["PRAGMA %s." % quote(schema)]
        else:
            # because PRAGMA looks in all attached databases if no schema
            # given, need to specify "main" schema, however since we want
            # 'temp' tables in the same namespace as 'main', need to run
            # the PRAGMA twice
            statements = ["PRAGMA main.", "PRAGMA temp."]
    
        qtable = quote(table_name)
        for statement in statements:
            statement = "%s%s(%s)" % (statement, pragma, qtable)
>           cursor = connection.exec_driver_sql(statement)
E           AttributeError: 'Engine' object has no attribute 'exec_driver_sql'

venv/lib/python3.7/site-packages/sqlalchemy/dialects/sqlite/base.py:2520: AttributeError
```

--- a/src/bluesearch/database/mining_cache.py
+++ b/src/bluesearch/database/mining_cache.py
@@ -279,7 +279,7 @@ class CreateMiningCache:
         self.logger = logging.getLogger(self.__class__.__name__)
         required_tables = ["articles", "sentences"]
         for table_name in required_tables:
-            if not database_engine.dialect.has_table(database_engine, table_name):
+            if not database_engine.has_table(table_name):
                 raise ValueError(
                     f"Database at {database_engine.url} does not "
                     f"contain required table {table_name}."
@@ -322,7 +322,7 @@ class CreateMiningCache:
         """Create the schemas of the different tables in the database."""
         metadata = sqlalchemy.MetaData()
 
-        if self.engine.dialect.has_table(self.engine, self.target_table):
+        if self.engine.has_table(self.target_table):
             self.mining_cache_table = sqlalchemy.Table(
                 self.target_table, metadata, autoload=True, autoload_with=self.engine
             )

